docs/rules/no-controllers.md

```diff
@@ -0,0 +1,32 @@
+# no-controllers
+
+Some people may prefer to avoid the use of controllers in their applications, typically in favor of components which can be more portable and easier to test.
+
+This rule disallows controller usage, except when the controller is used to define `queryParams`.
+
+## Rule Detail
+
+Examples of **incorrect** code for this rule:
+
+```js
+import Controller from '@ember/controller';
+export default class ArticlesController extends Controller {
+  // Controller disallowed since it doesn't use `queryParams`.
+  ...
+}
+```
```

Comment:
what about `Controller.extend({ ... })`?

if it's supported then it might be worth it to add an example for it too :)

Reply:
We may want to stop showing duplicate examples with both classic classes and native classes. I was thinking that we would only show native class examples going forward (while still supporting both). What do you think?

Reply:
there are still lots of apps around that don't use native classes yet. if we don't include a full example, we should at least mention that the rule supports the classic classes too.

Reply:
I added `Note that this rule supports both classic and native class syntax.` to the doc.

And in a separate PR, I'll try updating all the rule docs to mention their status regarding support for class and native class syntax.